checkov/arm/checks/resource/AppServiceAuthentication.py

```diff
@@ -1,36 +1,39 @@
+from __future__ import annotations
+
+from typing import Any
+
 from checkov.common.models.enums import CheckResult, CheckCategories
 from checkov.arm.base_resource_check import BaseResourceCheck
 
 # https://docs.microsoft.com/en-us/azure/templates/microsoft.web/2019-08-01/sites/config-authsettings
 
 
 class AppServiceAuthentication(BaseResourceCheck):
-    def __init__(self):
+    def __init__(self) -> None:
         name = "Ensure App Service Authentication is set on Azure App Service"
         id = "CKV_AZURE_13"
-        supported_resources = ['Microsoft.Web/sites/config', 'config']
-        categories = [CheckCategories.IAM]
+        supported_resources = ("Microsoft.Web/sites/config", "config")
+        categories = (CheckCategories.IAM,)
         super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
 
-    def scan_resource_conf(self, conf):
-        if "type" in conf:
-            if conf["type"] == "Microsoft.Web/sites/config":
-                if "name" in conf:
-                    if "authsettings" in conf["name"]:
+    def scan_resource_conf(self, conf: dict[str, Any]) -> CheckResult:
+        if self.entity_type == "Microsoft.Web/sites/config":
+            if "name" in conf and "authsettings" in conf["name"]:
+                if "properties" in conf and "enabled" in conf["properties"]:
+                    if str(conf["properties"]["enabled"]).lower() == "true":
+                        return CheckResult.PASSED
+                return CheckResult.FAILED
```

Comment:
Probably it's better to indent `return CheckResult.FAILED` to the same indentation level of `if "name" in conf and "authsettings" in conf["name"]:`

Reply:
yeah, you are totally right 😄